File: app/src/main/java/ru/ra66it/updaterforspotify/data/network/NetworkChecker.java
package ru.ra66it.updaterforspotify.data.network;

import android.content.Context;
import android.net.ConnectivityManager;
import android.net.NetworkInfo;

import ru.ra66it.updaterforspotify.UpdaterApp;

/**
 * Created by 2Rabbit on 21.09.2017.
 */

public class NetworkChecker {
    public static boolean isNetworkAvailable() {
        ConnectivityManager connectivityManager = (ConnectivityManager) UpdaterApp.getContext().getSystemService(Context.CONNECTIVITY_SERVICE);
        NetworkInfo activeNetworkInfo = connectivityManager.getActiveNetworkInfo();
        return activeNetworkInfo != null && activeNetworkInfo.isConnected();
    }
}

File: app/src/main/java/ru/ra66it/updaterforspotify/data/network/NetworkChecker.kt
package ru.ra66it.updaterforspotify.data.network

import android.content.Context
import android.net.ConnectivityManager
import ru.ra66it.updaterforspotify.UpdaterApp

object NetworkChecker {
    val isNetworkAvailable: Boolean
        get() {
            val connectivityManager =
                UpdaterApp.instance.getSystemService(Context.CONNECTIVITY_SERVICE) as ConnectivityManager
            val activeNetworkInfo = connectivityManager.activeNetworkInfo
            return activeNetworkInfo != null && activeNetworkInfo.isConnected
        }
}
